### PR TITLE
Make ip concrete

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: hase/symbex/tracer.py [hunk 1]
```diff
@@ -100,39 +133,42 @@ def __init__(
...
+        self.hook_plt_idx = list(self.hook_target.keys())
+        self.hook_plt_idx.sort()
+        self.filter.entry_check()
+        self.start_state = create_start_state(self.project, self.trace, self.cdanalyzer)
+        self.start_state.inspect.b(
+            "call", when=angr.BP_BEFORE, action=self.concretize_indirect_calls
+        )
+        res = self.start_state.inspect.b(
+            "successor", when=angr.BP_AFTER, action=self.concretize_state_ip
+        )
```
Comment (Mic92) [inline hase/symbex/tracer.py]: @Airtnp Maybe you can also have a look at the concretize functions.

**Comment by Mic92** [comment]
Multiple successors are ignored at the moment. They are
not so common and this is an optimization after all. From my understanding block guards should be still triggered by writing to `ip=` https://github.com/hase-project/angr/blob/angr-hacks-8.18.10.25/angr/sim_state.py#L202
and new constraint that access ip should receive the concrete value.
Am I am missing something else here.

**Comment by Airtnp** [comment]
Every branch will have multiple successors with guard (control flow constraints) carried with the successor (https://github.com/angr/angr/blob/ac28923eea519c681598158d976f7901ceb2b37d/angr/engines/successors.py#L150). So ignore multiple successors will give wrong constraints set.

Directly assign to ip will not trigger add constraints (https://github.com/angr/angr/blob/ac28923eea519c681598158d976f7901ceb2b37d/angr/engines/successors.py#L322)

**Comment by Mic92** [comment]
By ignoring I mean that it only sets the ip if there is a single successor: https://github.com/hase-project/hase/pull/65/files#diff-300213ee387938c49ef527bffedb3ce8R435

**Comment by Mic92** [comment]
Should be fixed in https://github.com/hase-project/hase/pull/68/files
However I still look for a test case to include in the test suite.

**Pull request merged**